library/init/meta/tactic.lean

```diff
@@ -73,6 +73,45 @@ meta def {u₁ u₂} tactic.down {α : Type u₂} (t : tactic (ulift.{u₁} α))
 | exception t ref s       := exception t ref s
 end
 
+namespace interactive
+
+/-- Typeclass for custom interaction monads, which provides
+    the information required to convert an interactive-mode
+    construction to a `tactic` which can actually be executed.
+
+    Given a `[monad m]`, `execute_with` explains how to turn a `begin ... end`
+    block, or a `by ...` statement into a `tactic α` which can actually be
+    executed. The `inhabited` first argument facilitates the passing of an
+    optional configuration parameter `config`, using the syntax:
+    ```
+    begin [custom_monad] with config,
+        ...
+    end
+    ```
+-/
+meta class executor (m : Type → Type u) [monad m] :=
+{config_type : Type}
```

Comment:
You shouldn't have `config_type` be implicit here. Conversely, does the implicitness of the `inhabited` instance allow it to be inferred in the `smt_config` example?

Reply:
@digama0 I'm pretty sure there's no `inhabited unit` in core?

Reply:
Oh, that has nothing to do with what you said, gotcha

Reply:
If there is no `inhabited unit` in core, then I know what the next PR is...

Reply:
Changed, but the instance can't be inferred in that example.

Reply:
@digama0 I thought about that, but there is a copy in mathlib (if I recall). So we run into a problem where we wouldn't be compatible between lean versions.